### PR TITLE
Update climate.py to support setting leavingWaterOffset

### PR DESCRIPTION
**Pull request by sraka1**
See https://github.com/speleolontra/daikin_residential_altherma/issues/179

Opening as a draft to validate approach...

@jwillemsen

### CODE REVIEW

File: custom_components/daikin_residential_altherma/climate.py [hunk 1]
```diff
@@ -119,6 +114,24 @@ def __init__(self, device):
                 self._supported_features |= SUPPORT_PRESET_MODE
             _LOGGER.info("Support_preset_mode {}: {}".format(mode,support_preset))
 
+    # In theory only one of the three should be settable
```
[the remark this answers is not in the document]
Comment (sraka1) [reply custom_components/daikin_residential_altherma/climate.py:117]: Hm. That's interesting. On code it defaults to the room temperature then being the setpoint. I think you'd need a separate ClimateEntity then for the leavingWaterOffset (as I suggested in the issue comment). However, I think as a stopgap, the PR at least enables systems where only the leavingWaterOffset is settable to be adjusted (right now it is not possible).

[the remark this answers is not in the document]
Comment (jwillemsen) [reply custom_components/daikin_residential_altherma/climate.py:117]: But this PR does make a lot of code more complex. What about a member to determine which set of data should be used (ATTR_TARGET_ROOM_TEMPERATURE or ATTR_TARGET_LEAVINGWATER_OFFSET or ATTR_TARGET_LEAVINGWATER_TEMPERATURE) and use that in all places. That would enable the creation of multiple ClimateEntities. Just store ATTR_TARGET_LEAVINGWATER_TEMPERATURE, not introduce another enum or flag.

[the remark this answers is not in the document]
Comment (jwillemsen) [reply custom_components/daikin_residential_altherma/climate.py:117]: This week I am pretty busy, maybe next week I have some time to try that approach

[the remark this answers is not in the document]
Comment (sraka1) [reply custom_components/daikin_residential_altherma/climate.py:117]: > What about a member to determine which set of data should be used (ATTR_TARGET_ROOM_TEMPERATURE or ATTR_TARGET_LEAVINGWATER_OFFSET or ATTR_TARGET_LEAVINGWATER_TEMPERATURE) and use that in all places. That would enable the creation of multiple ClimateEntities. Just store ATTR_TARGET_LEAVINGWATER_TEMPERATURE, not introduce another enum or flag.

Not sure what you mean by that exactly. P.S.: I am running this branch in my system and it works great (the gauge below is the leaving water offset)
Screenshot 2023-11-12 at 21 33 19

**Comment by edautz** [comment]
@sraka1. I replaced speleontra code by your code from your githubrepo in the custom directory in HA. My heatpump operates in roomTemperature mode. After restarting HA I don't see a way to control the leavingWaterOffset. Am I doing something wrong?

**Comment by jwillemsen** [comment]
Started with this, see https://github.com/jwillemsen/daikin_residential_altherma/tree/jwi-climatecontrolrework. This is really work in progress, only created multiple climate control entities, for each one found. When I have spare time next week I will continue with this.

**Comment by jwillemsen** [comment]
The code is very experimental, your issue is not fixed yet, would recommend a second HA setup to test this code

**Comment by edautz** [comment]
Did create a second HA docker instance and installed your code. I got two thermostat instances but as you had warned before the code is experimental and not fixed yet. The leaving water offset climate is not finished yet and got the same properties as room temperature climate entity.
 
image

image

**Comment by jwillemsen** [comment]
Thanks for the test, looks the logic correctly detects that there are two setpoints, need to get the correct values, probably monday I have some spare hours

**Comment by jwillemsen** [comment]
Extended to code to read the correct values, next step is to add support for controlling all of these

**Comment by jwillemsen** [comment]
Leaving water offset should now also work

**Comment by edautz** [comment]
Yes. You did it. Leaving water offset is HA controlable!!!! Very good.

Now I can build HA logic, to dynamicly control the offset and stay better on the target temperature. And create runs of weeks or months compensating the influence of rain, wind and sun.

What is your release plan of this branch in the production release of master branch?

**Comment by edautz** [comment]
A little remark:

image

Was in the old version:

image


Is it possible to change the name ClimateControl back to Altherma ?

**Comment by jwillemsen** [comment]
> A little remark:
> 
> image
> 
> Was in the old version:
> 
> image
> 
> Is it possible to change the name ClimateControl back to Altherma ?

That is not easy, with the new approach it can be that within one device there are multiple sensors with the same name (they exist in a group, for example "ClimateControl". This is one of the breaking changes, I want to finish first all rework and get rid of some more code, at that moment I will create a beta release to get more feedback

**Pull request closed without merging**